### DISCUSSION TITLE
Many bounding boxes appear with the same class name

### DISCUSSION THREAD

**Discussion by mridwanslubis** [ultralytics/yolov5#8030] [Q&A]
Hi, I think I have a problem with my yolov5 model.
I have 6 classes with 200 images in each class,  train with 16 batch and 100 epoch.
The result really good when testing with training data. But when testing with new data, so many bounding box appears with same class name, and with false detection results.

If you don't mind, please check the image down below to see what I mean.
![Screenshot from 2022-05-29 23-55-25](https://user-images.githubusercontent.com/52322511/170882929-f6368243-b0a6-4192-84fc-f63322ddfdfe.png)

This is the training results:

![Screenshot from 2022-05-30 00-25-57](https://user-images.githubusercontent.com/52322511/170883456-47c96d0c-b4a5-4146-9a6d-245aea1e8a9d.png)

By the way, I use the template code provided by Roboflow to do the training process, and don't change any code.
Did I miss something during the training process? and is there any configuration to adjust?

Any help, i'm very grateful.

**Comment by glenn-jocher** [comment]
@mridwanslubis your model suffers from poor generalization due to how you've created your dataset. For best results your deployment images and validation images must be similar.
![Screenshot 2021-08-04 at 13 15 05](https://user-images.githubusercontent.com/26833433/170884267-27e684df-ff39-4c66-952a-e0c52de88b0b.jpg)

**Reply by mridwanslubis** [reply]
@glenn-jocher thanks for the explanation sir. So did you mean the deploy images must be similiar with train and validation images? if met, will result in good object detection? is that what you mean sir?

![Screenshot from 2022-05-30 01-06-04](https://user-images.githubusercontent.com/52322511/170884975-d1556b7c-4f0a-4260-87af-b549f5a1d926.png)

Above is the result of detection using training data, and I think the only difference with deploy images is the background of the image.

**Reply by glenn-jocher** [reply]
Deployment space must be similar to val space for results to generalize.

**Reply by mridwanslubis** [reply]
@glenn-jocher  Actually I don't really understand what you mean, do you mean "deployment space" is the amount of testing data?
or if you dont mind, can you share an explanation link of the "problem dataset" image from ultralytics that you share? or other reading links that can give me understanding.
